flux-kube/plugins/flux-kube.c

```diff
@@ -0,0 +1,284 @@
+             flux_log_error (flux_jobtap_get_flux (p),
+                             "flux-kube: failed to post free event");
+    }
+    return 0;
+}
+
+static int validate_cb (flux_plugin_t *p,
+                        const char *topic,
+                        flux_plugin_arg_t *args,
+                        void *arg)
+{
+    int fluxkube = 0;
+    json_t *jobspec = NULL;
+    char *js, *s;
+    uint32_t userid = (uint32_t) -1;
+
+    if (flux_plugin_arg_unpack (args,
+                                FLUX_PLUGIN_ARG_IN,
+                                "{s:i s:{s:{s:{s?i}}}}",
+                                "userid", &userid,
+                                "jobspec",
+                                 "attributes",
+                                  "system",
+                                   "flux-kube", &fluxkube) < 0) {
+        return flux_jobtap_reject_job (p,
+                                       args,
+                                       "invalid system.flux-kube: %s",
+                                       flux_plugin_arg_strerror (args));
+    }
+
+    /*  Nothing to do if not fluxkube
+     */
+    if (!fluxkube)
+        return 0;
+
+    if (userid != getuid ())
+        return flux_jobtap_reject_job (p,
+                                       args,
+                                       "Guest user cannot use alloc bypass");
+
+    if (flux_plugin_arg_unpack (args,
+                                FLUX_PLUGIN_ARG_IN,
+                                "{s:o}",
+                                "jobspec", &jobspec) < 0) {
+        return flux_jobtap_reject_job (p,
+                                       args,
+                                       "invalid jobspec: %s",
+                                       flux_plugin_arg_strerror (args));
+    }
+   /*  Store jobspec in job structure to avoid re-fetching from plugin args
+     *   in job.state.sched callback.
+     */
+    if (!(js = json_dumps (jobspec, 0))
+        || flux_jobtap_job_aux_set (p,
+                                    FLUX_JOBTAP_CURRENT_JOB,
+                                    "flux-kube::jobspec",
+                                    js,
+                                    free) < 0) {
+        free (js);
+        return flux_jobtap_reject_job (p,
+                                       args,
+                                       "failed to capture flux-kube "
+                                       "jobspec: %s",
+                                       strerror (errno));
+    }
+    /*  Store R string in job structure to avoid re-fetching from plugin args
+     *   in job.state.sched callback.
+     */
+    s = "{\"version\": 1, "
+          "\"execution\": "
+            "{\"R_lite\": "
+              "[{\"rank\": \"0\", "
+                "\"children\": "
+                  "{\"core\": \"0\"}}], "
+              "\"starttime\": 0.0, "
+              "\"expiration\": 0.0, "
+              "\"nodelist\": "
+              "[\"kubernetes\"]}}";
```

Comment:
Does this work when rank 0 is running on a node not named `kubernetes`?

Reply:
Yes, this is just the dummy R string that is inserted into the KVS. We could make it more informative in the future, but it's largely a placeholder now.